src/sentry/static/sentry/app/components/sidebar/index.jsx

```diff
@@ -266,17 +267,26 @@ class Sidebar extends React.Component {
                 </Feature>
 
                 <Feature features={['events']} organization={organization}>
-                  <SidebarItem
-                    {...sidebarItemProps}
-                    onClick={(_id, evt) =>
-                      this.navigateWithGlobalSelection(
-                        `/organizations/${organization.slug}/events/`,
-                        evt
-                      )}
-                    icon={<InlineSvg src="icon-stack" />}
-                    label={t('Events')}
-                    to={`/organizations/${organization.slug}/events/`}
-                  />
+                  <StyledItem>
+                    <SidebarItem
```

Comment:
So, what I would actually like to do here is make `SidebarItem` use hooks inside of itself to determine what should be rendered as part of the item.

Right now this explicitly adds hooks into specific sidebar items that we know we want icons next to in getsentry. But that logic, knowing which sidebar items to put icons next to, is logic that only getsentry should be aware of. This is in the same spirit of having a generic name for the hook, we want to try and avoid adding in getsentry specific logic / functionality into sentry.

So what I think we could do is setup a hook that overrides the sidebar label, allowing customization of the label, which is more general functionality

https://github.com/getsentry/sentry/blob/0f09b27c2012f866e75aa59a56db1c86bad6f643/src/sentry/static/sentry/app/components/sidebar/sidebarItem.jsx#L91-L93

So here we can plug a hook in here and call it something like `sidebar:item-label`, which allows us to override the item label like so

```jsx
<Hook name="sidebar:item-label">
    {({hooks}) =>  hooks.map(h => h({
        organization,
        id: this.props.id // This is new, SidebarItems will have to specify ids.        
        label: (
                <SidebarItemLabel>
                    <TextOverflow>{label}</TextOverflow>
                </SidebarItemLabel>
            ),
        }))}
</Hook>
```

Or well, something close to that. This won't work 100% since if there's no hook defined it will return null, which means we'll just endup with empty labels on sentry.

Then in getsentry you can add a hook that will look at the label id and determine if it should add the power icon for it.

/cc @billyvg for thoughts

Reply:
Yeah sounds good - would `HookOrDefault` help here?

Reply:
Didn't know about `HookOrDefault`. 👍

Reply:
Yeah hook or default works well here for now but def agree that we could refactor `Hook` with the `default`